### PR TITLE
docs: add support for docs versioning (#1239)

### PR DESCRIPTION
**Pull request by kozub**
**Issue #, if available:**
#1239 

## Description of changes:

Add support for versioning:
- use Mike in mkdocs configuratin
- update github workflows (base on PowerTools Python and TypeScript config)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/on-doc-merge.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Docs
+
+on:
+  push:
+    branches:
+      - main
```
Comment (scottgerring) [inline .github/workflows/on-doc-merge.yml:6]: Should this run on push to main or release ?

File: .github/workflows/on-doc-merge.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Docs
+
+on:
+  push:
+    branches:
```
Comment (scottgerring) [inline .github/workflows/on-doc-merge.yml:5]: I think this should run on PR too

Comment (scottgerring) [reply .github/workflows/on-doc-merge.yml:5]: What we _might_ want to do on merge to main, is just build the documentation, to make sure we still can.

Comment (kozub) [reply .github/workflows/on-doc-merge.yml]: Hey @scottgerring  - I've updated configuration. Please have a look.

**Review by scottgerring — APPROVED** [review]
LGTM. I think we should uncomment the S3 publish and try trigger it manually for an old release from main, then patch if needed. Keen to get @sthulb 's feedback as well as he has worked a lot on this

**Comment by jeromevdl** [comment]
@scottgerring @kozub the doc is still not versioned (https://docs.powertools.aws.dev/lambda/java/). How do we do that?

**Comment by sthulb** [comment]
https://github.com/aws-powertools/powertools-lambda-java/actions/runs/5689418997

this PR contains errors

**Pull request merged**